nixos/modules/tasks/network-interfaces-scripted.nix

```diff
@@ -253,7 +253,11 @@ let
             '';
           };
 
-        createTunDevice = i: nameValuePair "${i.name}-netdev"
+          createTunDevice = i: nameValuePair "${i.name}-netdev"
+          ( let
+              greType = if i.virtualType == "gretun" then "gre" else i.virtualType;
+              greLocal = optionalString (i.gre.local != null) " local ${i.gre.local}";
```

Comment:
this entire block could be moved to the `script` attribute that uses it. would probably be easier to read like that, adding lets like this is rather uncommon

Reply:
Yep, that clears things up. Done.